pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -176,16 +176,23 @@ func (c *Controller) Apply(change *sink.Change) error {
 	}
 
 	var prevStore map[string]map[string]*model.Config
-
 	c.configStoreMu.Lock()
 	prevStore = c.configStore[descriptor.Type]
 	c.configStore[descriptor.Type] = innerStore
 	c.configStoreMu.Unlock()
 
-	if descriptor.Type == model.ServiceEntry.Type {
+	if c.options.IncrementalEDS &&
+		descriptor.Collection == model.ServiceEntry.Collection ||
+		descriptor.Collection == model.SyntheticServiceEntry.Collection {
+		if err := c.incrementalUpdate(innerStore, prevStore); err != nil {
```

Comment:
@costinm IIRC, you wanted to do incremental only if endpoints were changed ... if the service part changed, we would still do a full update.  Can you confirm?

Reply:
That might not be what is happening here, this currently only checks if the current config version is changed from the previous config and then it does the incremental update. I will update to match that logic, just to confirm if endpoints changed and the service is the same do incremental updates of the endpoints. If the endpoints are same and service is changed then do a full update. Also if both service and endpoint changed do a full update. Does this logic capture what with want to do with the incremental EDS?  @nmittler @costinm